backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java

```diff
@@ -158,13 +177,27 @@ private void initializeS3ClientWithTemporaryCredentials(Regions clientRegion) {
 
   private void RefreshCredentialsAndSchedule(String region) {
     LOGGER.debug("fetching token for s3 access");
+    if (task != null) {
+      task.cancel();
+      task = null;
+    }
 
     LOGGER.debug(
         "credentials before refresh {}",
         temporarySessionCredentials != null ? temporarySessionCredentials.hashCode() : null);
 
-    TimerTask task = new FetchTemporaryS3Token(region);
-    task.run();
+    scheduleTimer(region);
+  }
+
+  public static synchronized void scheduleTimer(String region) {
```

Comment:
comment for conrado's review: synchronized to avoid multiple thread doing the operations in parallel